plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -0,0 +1,156 @@
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.util.AsyncQueue;
+import io.trino.plugin.hive.util.ThrottledAsyncQueue;
+import io.trino.plugin.hudi.query.HudiFileLister;
+import io.trino.plugin.hudi.query.HudiQueryMode;
+import io.trino.plugin.hudi.split.HudiBackgroundSplitLoader;
+import io.trino.plugin.hudi.split.HudiSplitWeightProvider;
+import io.trino.plugin.hudi.split.SizeBasedSplitWeightProvider;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorSplitSource;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.engine.HoodieEngineContext;
+import org.apache.hudi.common.engine.HoodieLocalEngineContext;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.concurrent.MoreFutures.toCompletableFuture;
+import static io.trino.plugin.hudi.HudiSessionProperties.getMinimumAssignedSplitWeight;
+import static io.trino.plugin.hudi.HudiSessionProperties.getStandardSplitWeightSize;
+import static io.trino.plugin.hudi.HudiSessionProperties.isHudiMetadataEnabled;
+import static io.trino.plugin.hudi.HudiSessionProperties.isSizeBasedSplitWeightsEnabled;
+import static io.trino.plugin.hudi.HudiSessionProperties.shouldUseMetastoreForPartitions;
+import static io.trino.plugin.hudi.query.HudiFileListerFactory.create;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.util.stream.Collectors.toList;
+
+public class HudiSplitSource
+        implements ConnectorSplitSource
+{
+    private final AsyncQueue<ConnectorSplit> queue;
+    private final AtomicReference<TrinoException> trinoException = new AtomicReference<>();
+
+    public HudiSplitSource(
+            ConnectorSession session,
+            HiveMetastore metastore,
+            Table table,
+            HudiTableHandle tableHandle,
+            FileSystem fs,
+            Map<String, HiveColumnHandle> partitionColumnHandleMap,
+            ExecutorService executor,
+            int maxSplitsPerSecond,
+            int maxOutstandingSplits)
+    {
+        boolean metadataEnabled = isHudiMetadataEnabled(session);
+        boolean useMetastoreForPartitions = shouldUseMetastoreForPartitions(session);
+        HoodieTableMetaClient metaClient = buildTableMetaClient(fs.getConf(), tableHandle.getBasePath());
+        HoodieEngineContext engineContext = new HoodieLocalEngineContext(fs.getConf());
```

Comment:
Does Hudi interact directly with the Hadoop file system internally?  If so, that will not work for us.  We are moving away from using Hadoop and Hive apis in Trino directly and are replacing these with custom code.  We are already pretty far decoupling Iceberg and Delta using the https://github.com/trinodb/trino/tree/master/lib/trino-filesystem/src/main/java/io/trino/filesystem code.  We need a way to not use these libraries at all, so I'm not sure we want to merge this code until we have a path forward on this.  I say that because we already have the path for everything else using this, and I don't want us to get trapped on these systems at the last minute becaue Hudi got added... Please take a look at the Iceberg file system abstractions for inspiration on how to do this cleanly.

Instead if this is a way to configure the system, please switch this to using standard Trino/Airlift configuration values.  The Hadoop configurations are only used for HDFS (because it is the only way... one of the reasons we are moving away)